draft-ietf-tls-certificate-compression.md

```diff
@@ -64,13 +65,18 @@ meaning defined in [RFC2119].
 
 # Negotiating Certificate Compression
 
-This document defines a new extension type (compress_certificates(TBD)), which
-is used by the client and the server to negotiate the use of compression for
-their certificate chains, as well as the choice of the compression algorithm.
-
-By sending the compress_certificates extension, the client indicates to the
-server the certificate compression algorithms it supports.  The "extension_data"
-field of this extension in the ClientHello SHALL contain a
+This document defines a new extension type (compress_certificate(TBD)), which
+can be used to signal the supported compression formats for the Certificate
+message to the peer.  Whenever it is sent by the client as a ClientHello message
+extension ([RFC5246], Section 7.4.1.4), it indicates the support for compressed
+server certificates.  Whenever it is sent by the server as a CertificateRequest
+extension ([I-D.ietf-tls-tls13], Section 4.3.2), it indicates the support for
+compressed client certificates.  Note that CertificareRequest messages are not
+supported by TLS versions prior to 1.3.
```

Comment:
So prior to TLS 1.3, how does the client use this?  The signature_algorithms extension in 1.3 allows the server to advertise it (in EncryptedExtensions of course) so that the client can know.  Do you think that's valuable?  Obviously this is just cleaner all around, except for the 1.2 client certificate problem, which you could reasonably conclude isn't worth worrying about.

Reply:
I am not sure this is particularly valuable, which is why I just went straight for a more elegant version.